jgiven-core/src/test/java/com/tngtech/jgiven/report/asciidoc/AsciiDocReportModelVisitorTest.java

```diff
@@ -0,0 +1,296 @@
+package com.tngtech.jgiven.report.asciidoc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.tngtech.jgiven.report.ReportBlockConverter;
+import com.tngtech.jgiven.report.model.CasesTable;
+import com.tngtech.jgiven.report.model.ExecutionStatus;
+import com.tngtech.jgiven.report.model.ReportModel;
+import com.tngtech.jgiven.report.model.ReportStatistics;
+import com.tngtech.jgiven.report.model.ScenarioCaseModel;
+import com.tngtech.jgiven.report.model.ScenarioModel;
+import com.tngtech.jgiven.report.model.StepModel;
+import com.tngtech.jgiven.report.model.StepStatus;
+import com.tngtech.jgiven.report.model.Word;
+import java.util.List;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AsciiDocReportModelVisitorTest {
+    private final ReportBlockConverter blockConverter = new MyFakeReportBlockConverter();
+
+    private AsciiDocReportModelVisitor reportModelVisitor;
+
+    @Before
+    public void setUp() {
+        ReportStatistics reportStatistics = new ReportStatistics();
+        reportModelVisitor = new AsciiDocReportModelVisitor(blockConverter, reportStatistics);
+    }
+
+    @Test
+    public void visits_a_simple_report() {
+        // given
+        ReportModel report = mkReport(mkScenario("Simple Scenario", false, mkScenarioCase(
+                mkStep("Given", "state"),
+                mkStep("When", "action"),
+                mkStep("Then", "outcome"))));
+
+        // when
+        report.accept(reportModelVisitor);
+
+        // then
+        assertThat(reportModelVisitor.getResult())
+                .isEqualTo(ImmutableList.of(
+                        "FeatureHeaderBlock",
+                        "ScenarioHeaderBlock",
+                        "FirstStepBlock",
+                        "StepBlock",
+                        "StepBlock",
+                        "ScenarioFooterBlock"));
+    }
+
+    @Test
+    public void visits_a_report_with_two_scenarios() {
+        // given
+
+        ReportModel report = mkReport(
+                mkScenario("Scenario One", false, mkScenarioCase(
```

Comment:
Its really rare to see helper /shortcut methods in a test that work that well. Nice :)